src/components/Modal/index.js

```diff
@@ -7,15 +7,19 @@ import './styles.css'
 
 const ModalComponent = forwardRef(({children, additionalClass}, ref) => {
     const [modalStatus, setModalStatus] = useState(false);
+    console.log(modalStatus)
 
-    useImperativeHandle(ref, () => ({
-        openModal() {
-            setModalStatus(true);
-        },
-        closeModal() {
-            setModalStatus(false);
+    useImperativeHandle(ref, () => {
+        return {
+            openModal: () => {
+                console.log("setting true")
+                setModalStatus(true);
+            },
+            closeModal: () => {
+                setModalStatus(false);
+            }
         }
-    }));
+    }, [ref]);
```

Comment:
`useImperativeHandle` já é feito para trabalhar com refs, a dependência no hook não é necessária. Segundo o lint:
"React Hook useImperativeHandle has an unnecessary dependency: 'ref'. Either exclude it or remove the dependency array"
```suggestion
    }, []);
```